notation/sign.ts

```diff
@@ -0,0 +1,96 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as taskLib from 'azure-pipelines-task-lib/task';
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+
+import { getDownloadInfo, installFromURL } from './lib/install';
+
+import { getArtifactReferences } from './lib/variables';
+import { getConfigHome } from './lib/fs';
+import { getVaultCredentials } from './lib/credentials';
+
+const AZURE_KV_PLUGIN_VERSION_FILE = 'azure_kv_versions.json';
+const AZURE_KV_VERSION_LOCK_FILE = 'azure_kv_version_lock.json';
+
+export async function sign(): Promise<void> {
+    const artifactRefs = getArtifactReferences();
+    const keyid = taskLib.getInput('keyid', true) || '';
+    const cacerts = taskLib.getInput('cacerts', false) || '';
+    const selfSignedCert = taskLib.getBoolInput('selfSigned', false) === true;
+    const signatureFormat = taskLib.getInput('signatureFormat', false) || 'cose';
+    const allowReferrerAPI = taskLib.getBoolInput('allowReferrersAPI', false);
+    const debug = taskLib.getVariable('system.debug')
+
+    await installPlugin();
+
+    let env = { ...process.env, ...await getVaultCredentials() }
+    if (allowReferrerAPI) {
+        env["NOTATION_EXPERIMENTAL"] = "1"
+    }
```

Comment:
`NOTATION_EXPERIMENTAL` is a switch that should only be controlled by the user. They need to aware that `allowReferrerAPI` is an experimental feature in Notation. If they want to use it, they have to explicitly set the env in their pipeline. Implicitly setting this env in the code does not align with the meaning of `NOTATION_EXPERIMENTAL`.

Reply:
For GUI, user can see there is a `[expreimental]` mark:
![image](https://github.com/AzureCR/notation-ado-task/assets/43160897/8423bfd1-62b5-47c3-aeae-9c45eefe36aa)

For doc, user can see the filed is marked as expreimental.

So, I just want to simplify user experience. Set env is not convenient for pipeline. Also we can talk with PMs about it.